### PR TITLE
Operator Overloading in C++

### PR DESCRIPTION
**Pull request by Akriti-Choudhary**
<hr>

## Description 📜

The documentation provides details about Operator Overloading in C++ along with some codes.

Fixes # <issue_number_here> #1754 

<hr>

## Type of change 📝

- [ x] Documentation (Content Creation in the form of codes or tutorials)

<hr>

## Domain of Contribution 📊

- [x ] C/CPP

<hr>

## Checklist ✅

- [ x] I follow [Contributing Guidelines](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CONTRIBUTING.md) & [Code of conduct](https://github.com/girlscript/winter-of-contributing/blob/main/.github/CODE_OF_CONDUCT.md) of this project.
- [x ] I have performed a self-review of my own code or work.
- [x ] I have commented on my code, particularly in hard-to-understand areas.
- [x ] My changes generates no new warnings.
- [x ] I'm GWOC'21 contributor

<hr>

### CODE REVIEW

File: C_CPP/Operator Overloading/operator_overloading_in_c++.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+Displaying the elements of the array :        
+90 99 78 120 187
+```
+
+```c
+**Code for operator overloading implemented using non-member function**
+
+/*Add two arrays using operator overloading implemented through non-member (friend) function*/
+
+#include <iostream>
```
Comment (vasu-1) [inline C_CPP/Operator Overloading/operator_overloading_in_c++.md:126]:
```suggestion

**Code for operator overloading implemented using non-member function**

```c
/*Add two arrays using operator overloading implemented through non-member (friend) function*/

#include <iostream>
```

File: C_CPP/Operator Overloading/operator_overloading_in_c++.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    AddArray();                     //defaut constructor
+    AddArray(int);                  //paremeterized constructor
+    void display();                 //function for displaying the elements of an array
+    friend AddArray operator+(AddArray ,AddArray ); //non-member function to overload + operator
+};
+
+AddArray::AddArray()
+{
+    for (int i = 0; i < 5; ++i)
+    {
```
Comment (vasu-1) [inline C_CPP/Operator Overloading/operator_overloading_in_c++.md:146]:
```suggestion
public:
    AddArray();                     //default constructor
    AddArray(int);                  //parameterized constructor
    void display();                 //function for displaying the elements of an array
    friend AddArray operator+(AddArray ,AddArray ); //non-member function to overload +operator
};

AddArray::AddArray()
{
    for (int i = 0; i < 5; ++i)
    {
```

**Review by vasu-1 — CHANGES_REQUESTED** [review]
Please do this changes !

File: C_CPP/Operator Overloading/operator_overloading_in_c++.md [hunk 3]
```diff
@@ -1,4 +1,4 @@
-**Operator Overloading**
+#Operator Overloading
```
Comment (vasu-1) [inline C_CPP/Operator Overloading/operator_overloading_in_c++.md]:
```suggestion
# Operator Overloading

```

add this space

**Review by vasu-1 — CHANGES_REQUESTED** [review]

**Comment by Akriti-Choudhary** [comment]
@vasu-1  I have done all the required changes. Please, review the project again.

**Review by Suvraneel — COMMENTED** [review]
Also, add a line at Operator Overloading/readme.md

**Comment by Suvraneel** [comment]
@vasu-1 Nice review btw 💯

**Comment by Akriti-Choudhary** [comment]
@Suvraneel I have updated the readme via PR https://github.com/girlscript/winter-of-contributing/pull/2367
Kindly, Review this.

**Review by vasu-1 — APPROVED** [review]
Done

**Comment by Suvraneel** [comment]
@Akriti-Chaudhary Also, add a line at Operator Overloading/readme.md in this patch-2 branch itself

**Comment by Akriti-Choudhary** [comment]
@Suvraneel  alright, done.

**Comment by Suvraneel** [comment]
Way to go @Akriti-Choudhary ... ❤️  Great :octocat:

**Comment by Akriti-Choudhary** [comment]
Thank you @Suvraneel and @vasu-1 for correcting me constantly.

**Pull request merged**